crates/net/network/src/fetch/mod.rs

```diff
@@ -483,4 +483,34 @@ mod tests {
         assert_eq!(fetcher.next_peer(), Some(peer2));
         assert_eq!(fetcher.next_peer(), Some(peer2));
     }
+
+    #[tokio::test]
+    async fn test_on_block_headers_response() {
+        let manager = PeersManager::new(PeersConfig::default());
+        let mut fetcher = StateFetcher::new(manager.handle(), Default::default());
+        let peer_id = H512::random();
+
+        assert_eq!(fetcher.on_block_headers_response(peer_id, Ok(vec![Header::default()])), None);
+
+        assert_eq!(
+            fetcher.on_block_headers_response(peer_id, Err(RequestError::Timeout)),
+            Some(BlockResponseOutcome::BadResponse(peer_id, ReputationChangeKind::Timeout))
+        );
+        assert_eq!(
+            fetcher.on_block_headers_response(peer_id, Err(RequestError::BadResponse)),
+            None
+        );
+        assert_eq!(
+            fetcher.on_block_headers_response(peer_id, Err(RequestError::ChannelClosed)),
+            None
+        );
+        assert_eq!(
+            fetcher.on_block_headers_response(peer_id, Err(RequestError::ConnectionDropped)),
+            None
+        );
+        assert_eq!(
+            fetcher.on_block_headers_response(peer_id, Err(RequestError::UnsupportedCapability)),
+            None
+        );
```

Comment:
Is this right? Wouldn't these return `BlockResponseOutcome::BadResponse` before, and now they return `None`? Seems weird?

Reply:
As @mattsse [stated above](https://github.com/paradigmxyz/reth/pull/1086#pullrequestreview-1275310529), most of them are handled by the active session (Sorry, I've marked all his comments as solved).

Reply:
this variant is actually not used here because we ensure a session has eth support.

we could remove this variant entirely. atm it would only be relevant if there's an incoming request for capability other than `eth`

Reply:
I'll add some additional docs on reporting